var/spack/repos/builtin/packages/intel-oneapi-runtime/package.py

```diff
@@ -0,0 +1,52 @@
+# Copyright 2013-2024 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+import os
+
+from llnl.util import tty
+
+from spack.package import *
+from spack.pkg.builtin.gcc_runtime import get_elf_libraries
+
+
+class IntelOneapiRuntime(Package):
+    """Package for OneAPI compiler runtime libraries"""
+
+    homepage = "https://software.intel.com/content/www/us/en/develop/tools/oneapi.html"
+    has_code = False
+
+    tags = ["runtime"]
+
+    requires("%oneapi")
+
+    depends_on("gcc-runtime", type="link")
+
+    LIBRARIES = [
+        "imf",
+        "intlc",
+        "irng",
+        "svml",
+        "ifcore",  # Fortran
+        "ifcoremt",  # Fortran
+        "ifport",  # Fortran
+        "sycl",
+    ]
+
+    # libifcore ABI
+    provides("fortran-rt", "ifcore@5", when="%oneapi@2021:")
+
+    def install(self, spec, prefix):
+        if spec.platform in ["linux", "cray", "freebsd"]:
+            libraries = get_elf_libraries(compiler=self.compiler, libraries=self.LIBRARIES)
+        else:
+            raise RuntimeError("Unsupported platform")
+
+        mkdir(prefix.lib)
+
+        if not libraries:
+            tty.warn("Could not detect any shared OneAPI runtime libraries")
+            return
+
+        for path, name in libraries:
```

Comment:
Instead of doing the symlink, the oneapi documentation tells people to make the prefix 2 levels up: https://spack.readthedocs.io/en/latest/build_systems/inteloneapipackage.html#libraries
and the packages know how to find the bin/lib/include 2 levels down. I tried introducing symlinks a while ago, but got rid of them for 2024 because of some conflicts (I only remember that it was MPI, not the details). If I was starting from nothing, I would introduce the symlinks on day 1.

This is a new package, so symlinks seems ok, and people will have to know that the external package prefix is not 2 levels up.

Reply:
@rscohn2 For other packages, like `gcc-runtime` we would actually copy the libraries in place. I didn't do it here because I think it would be a license infringement for OneAPI (if e.g. nodes are redistributed via `buildcache push`), whereas a symlink would not be an issue. Is that correct, or do you have any better suggestion?

Reply:
The library runtimes have a separate license and allow redistribution.

Reply:
nice

Reply:
@rscohn2 That's what we were doing initially, before this comment of yours :point_up: I'm a bit confused at the moment on what we can do... :sweat_smile:

Reply:
Todd suggested checking the redist to make sure the files were listed. When I did that, I saw that redistribution has obligations, like disallowing reverse engineering. It is ok to symlink, it is ok to copy, but it would be complicated to redistribute.  I had forgotten about the discussion above, but now I see that my answer was not a good one. Sorry if this caused you to do unnecessary work. Do you think there is a benefit to going back to symlinks vs marking the package as non-cacheable?

Reply:
> I saw that redistribution has obligations, like disallowing reverse engineering

@rscohn2: can you say why this complicates redistribution?  Seems like redistribution is allowed, and we expect users to comply with the licenses for packages. What's the extra complication here (other than another condition of the license)?

Reply:
This is the scenario the redistribution license was intended to support:

I build an application with icpx. I distribute the application with some intel libraries (e.g. libsycl, libimf, ...) I attach a license to my distribution. I need to make sure my use of the libraries complies with intel's license. 

This includes restricting the rights of people who use my distribution (e.g. no reverse engineering of intel libraries). My distribution needs to have a license that passes on the intel restrictions. This is doable. But it seems better to disable redistribution of the package.

Reply:
One of our goals is to allow users to easily build containers that use the runtime components of the intel compilers, and to keep the containers reasonably sized.  I think we're going to have a much harder time with the second part of that if we have to bundle the entire compiler vs. just the runtime.  So I think the goal here is to support the use case as described.

I'm not sure how redistributing vs. not redistributing makes a difference in users' ability to reverse engineer the libraries. They could download these libraries from intel if they wanted to reverse engineer, right?  The only difference is that they'd be getting the runtime libraries in a spack package.  If we need to talk to Intel lawyers to get on the same page with them we could do that -- e.g., UO has an arrangement that allows them to redistribute the entire intel compiler package (which we do not -- we don't put the intel packages in our mirror).